resources/puppetlabs/lein-ezbake/template/global/ext/cli/reload.erb

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+restartfile=/opt/puppetlabs/server/data/<%= EZBake::Config[:real_name] %>/restartcounter.txt
+cur=$(head -1 $restartfile)
+initial=$cur
+timeout="300"
+
+if [ -e $restartfile ];  then
+    kill -HUP $(pgrep -f <%= EZBake::Config[:uberjar_name] %>)
+    while [ "$cur" == "$initial" ] ;do
+        sleep 0.1
+        cur=$(head -1 $restartfile)
+
+        ((timeout--))
+        if [ "$timeout" = 0 ]; then
+            exit 1
+        fi
+    done
+else
+    echo "Restart file does not exist"
+    exit 1
```

Comment:
If we somehow ended up in a state where a project was consuming this version of ezbake but did *not* opt-in to the Trapperkeeper 'restart-file' feature, then the command (e.g. `puppetserver reload`) would always end up logging this error.  I *think* that probably seems like the right way to go, though we could change the error message to something more generic, like `Application 'puppetserver' does not support 'reload'.`, since we don't want users to ever need to know anything about these restart files.

The other option would be to have the script go ahead and send the HUP if the restart file doesn't exist, but then just exit immediately rather than waiting for the server to come up to ready, since it won't have a way of knowing about that.  I think I prefer the hard fail but am interested in hearing opinions from @underscorgan @shrug @haus or whomever else.